common/predictive-text/worker/index.ts

```diff
@@ -33,56 +116,84 @@ class LMLayerWorker {
    * so that this can be tested **outside of a Worker**.
    */
   private _postMessage: PostMessage;
-  private _importScripts: ImportScripts;
 
   constructor(options = {
     postMessage: null,
-    importScripts: null
   }) {
     this._postMessage = options.postMessage || postMessage;
-    this._importScripts = options.importScripts || importScripts;
   }
 
   /**
    * A function that can be set as self.onmessage (the Worker
    * message handler).
    * NOTE! You must bind it to a specific instance, e.g.:
-   * 
+   *
    *   // Do this!
    *   self.onmessage = worker.onMessage.bind(worker);
-   * 
+   *
    * Incorrect:
-   * 
+   *
    *   // Don't do this!
    *   self.onmessage = worker.onMessage;
-   * 
+   *
    * See: .install();
    */
   onMessage(event: MessageEvent) {
     const {message} = event.data;
+    // We must have gotten a message!
     if (!message) {
       throw new Error(`Missing required 'message' attribute: ${event.data}`)
     }
 
-    // Load the model.
-    let model = this._importScripts(event.data.model);
+    // ...that message must have been 'initialize'!
+    if (message !== 'initialize') {
+      throw new Error(`invalid message; expected 'initialize' but got ${message}`);
+    }
+
+    let payload: InitializeMessage = event.data;
+
+    let {model, configuration} = this.loadModel(
+      // TODO: validate configuration, and provide valid configuration in tests.
+      // @ts-ignore
+      payload.model, payload.configuration || {}
```

Comment:
Waaaait, noting stuff later on, `payload.model` would be a string of raw model code, not a filename/URI for use with loadScript?

Also, just to make sure I'm clear... your use of "model" here refers to a particular language's trained data for use in a FST or wordlist (In machine learning, the learned "parameters"), or does this refer to the function library used to process that data (the traditional machine learning definition of "model")?

Reply:
In this version, it's a string of raw source code. I need to try if the `Blob` works across Worker boundaries. If that's the case, I'll revert the removal of `importScripts()` and get rid of this `new Function('configuration', '...')` eval stuff. If it works, _then_ the model would be a URI that can be imported.

My definition of "model" is: code that implements the prediction interface. This is intended to be a _trained_ model bundled with the code to make it work. I can use a different word than `model` so we don't overload the term. Any suggestions?

Reply:
I'm not entirely clear yet, so maybe a specific example might help me.  If we have 10 different languages that implement FST-based predictive modeling, what you're saying means that none of them would share common code?  No common FST lookup function library or anything?

Reply:
That's correct. In this PR, I've assumed no common library within the Worker. However, now that I've thought about it, I think it's better if we provide a library for wordlists and FSTs. See #1335.